pkgs/tools/security/secp256k1/default.nix

```diff
@@ -1,32 +1,51 @@
-{ stdenv, fetchFromGitHub, autoconf, automake, libtool, ... }:
+{ stdenv, fetchFromGitHub, autoreconfHook, jdk
+
+# Enable ECDSA pubkey recovery module
+, enableRecovery ? true
+
+# Enable ECDH shared secret computation (disabled by default because it is
+# experimental)
+, enableECDH ? false
+
+# Enable libsecp256k1_jni (disabled by default because it requires a jdk,
+# which is a large dependency)
+, enableJNI ? false
+
+, ... }:
+
+let inherit (stdenv.lib) optionals; in
 
 stdenv.mkDerivation rec {
   name = "secp256k1-${version}";
 
-  # I can't find any version numbers, so we're just using the date
-  # of the last commit.
-  version = "2016-05-30";
+  # I can't find any version numbers, so we're just using the date of the
+  # last commit.
+  version = "2016-11-27";
 
   src = fetchFromGitHub {
     owner = "bitcoin-core";
     repo = "secp256k1";
-    rev = "b3be8521e694eaf45dd29baea035055183c42fe2";
-    sha256 = "1pgsy72w87yxbiqn96hnm8alsfx3rj7d9jlzdsypyf6i1rf6w4bq";
+    rev = "2928420c1b8e1feee8c20dff4e3cc41a0de2fc22";
+    sha256 = "1djsr2vrhh88353czlwb8bwlyabf008w1f7xg0fs3q33rf42w5gm";
   };
 
-  buildInputs = [ autoconf automake libtool ];
+  buildInputs = optionals enableJNI [ jdk ];
+
+  nativeBuildInputs = [ autoreconfHook ];
 
-  configureFlags = [ "--enable-module-recovery" ];
+  configureFlags =
+    optionals enableECDH [ "--enable-module-ecdh" "--enable-experimental" ] ++
+    optionals enableRecovery [ "--enable-module-recovery" ] ++
+    optionals enableJNI [ "--enable-jni" ];
 
   preConfigure = "./autogen.sh";
```

Comment:
You can remove this hook because `autoreconfHook` already does this in a new build phase.